lib/src/Pos/Logic/Types.hs

```diff
@@ -104,6 +106,9 @@ data Logic m = Logic
     , recoveryInProgress :: m Bool
 
     , securityParams     :: SecurityParams
+
+      -- action which runs whenever connection status changes
+    , connectionChangeAction :: ConnectionChangeAction m NodeId
```

Comment:
This doesn't fit with the logic layer. Instead of a callback-based approach, could we have the diffusion layer offer a `TVar ConnectionStatus`? Then the *control* part of the program (which uses logic and diffusion) can grab it and use it to implement the UI.